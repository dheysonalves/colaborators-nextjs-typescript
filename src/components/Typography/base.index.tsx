import styled from '@emotion/styled'
import { css } from '@emotion/react'
import { theme } from '../../styles/theme/lightTheme'
import DynamicTypography from './index'

export const MainStyle = css`
	font-size: 1.5rem;
`

const BaseTypography = styled(DynamicTypography)`
	${MainStyle};
	color: ${theme.light.neutral.black};
	font-weight: 600;
`

const Paragraph = styled(DynamicTypography)`
	${MainStyle};
	font-size: 1rem;
	color: ${theme.light.neutral.neutral_5};
	font-weight: 400;
`

export { BaseTypography, Paragraph }
