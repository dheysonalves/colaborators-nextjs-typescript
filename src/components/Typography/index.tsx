import React from 'react'
import styled from '@emotion/styled'

type TypographyProps = {
	tag: 'h1' | 'h2' | 'h3' | 'p'
	children: JSX.Element
}
const StyledDynamicTypography = styled.div``

const Typography = ({ tag = 'h1', children, ...props }: TypographyProps) => {
	const WithComponent = StyledDynamicTypography.withComponent(tag)

	return <WithComponent {...props}>{children}</WithComponent>
}

export default Typography
