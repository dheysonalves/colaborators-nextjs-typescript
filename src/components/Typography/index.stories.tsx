import { ComponentMeta, ComponentStory } from '@storybook/react'
import Typography from '.'

export default {
	title: 'Typography',
	component: Typography,
	args: {
		tag: 'h1',
		children: (
			<>
				<h1>Typography H1</h1>
			</>
		)
	}
} as ComponentMeta<typeof Typography>

export const Default: ComponentStory<typeof Typography> = (args) => (
	<Typography {...args} />
)
