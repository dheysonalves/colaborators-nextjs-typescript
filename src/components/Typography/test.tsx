import { render, screen } from '@testing-library/react'

import Typography from '.'

describe('<Typography />', () => {
	it('should render the heading', () => {
		const { container } = render(
			<Typography tag={'h1'}>Typography H1</Typography>
		)

		expect(
			screen.getByRole('heading', { name: /Typography H1/i })
		).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})

	it('should render the heading 2', () => {
		const { container } = render(
			<Typography tag={'h2'}>Typography H2</Typography>
		)

		expect(
			screen.getByRole('heading', { name: /Typography H2/i })
		).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})

	it('should render the heading 3', () => {
		const { container } = render(
			<Typography tag={'h3'}>Typography H3</Typography>
		)

		expect(
			screen.getByRole('heading', { name: /Typography H3/i })
		).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})

	it('should render the paragraph', () => {
		const { container } = render(
			<Typography tag={'p'}>Typography p</Typography>
		)

		expect(screen.getByText('Typography p')).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})
})
