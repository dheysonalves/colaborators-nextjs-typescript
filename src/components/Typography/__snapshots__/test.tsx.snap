// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> should render the heading 1`] = `
<h1
  class="css-0"
>
  Typography H1
</h1>
`

exports[`<Typography /> should render the heading 2 1`] = `
<h2
  class="css-0"
>
  Typography H2
</h2>
`

exports[`<Typography /> should render the heading 3 1`] = `
<h3
  class="css-0"
>
  Typography H3
</h3>
`

exports[`<Typography /> should render the paragraph 1`] = `
<p
  class="css-0"
>
  Typography p
</p>
`
